spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/procedures/RewriteDataFilesProcedure.java

```diff
@@ -0,0 +1,192 @@
+
+    return modifyIcebergTable(tableIdent, table -> {
+      RewriteDataFiles action = actions().rewriteDataFiles(table);
+
+      String strategy = args.isNullAt(1) ? null : args.getString(1);
+      SortOrder sortOrder = collectSortOrders(table, args.isNullAt(2) ? null : args.getString(2));
+      action = checkAndApplyStrategy(action, strategy, sortOrder);
+
+      if (!args.isNullAt(3)) {
+        action = checkAndApplyOptions(args, action);
+      }
+
+      String where = args.isNullAt(4) ? null : args.getString(4);
+      action = checkAndApplyFilter(action, where);
+
+      RewriteDataFiles.Result result = action.execute();
+
+      return toOutputRows(result);
+    });
+  }
+
+  private RewriteDataFiles checkAndApplyFilter(RewriteDataFiles action, String where) {
+    if (where != null) {
+      ParserInterface sqlParser = spark().sessionState().sqlParser();
+      try {
+        Expression expression = sqlParser.parseExpression(where);
+        return action.filter(SparkExpressionConverter.convertToIcebergExpression(expression));
+      } catch (ParseException e) {
+        throw new IllegalArgumentException("Cannot parse predicates in where option: " + where);
+      }
+    }
+    return action;
+  }
+
+  private RewriteDataFiles checkAndApplyOptions(InternalRow args, RewriteDataFiles action) {
+    Map<String, String> options = Maps.newHashMap();
+    args.getMap(3).foreach(DataTypes.StringType, DataTypes.StringType,
+            (k, v) -> {
+              options.put(k.toString(), v.toString());
+              return BoxedUnit.UNIT;
+            });
+    return action.options(options);
+  }
+
+  private RewriteDataFiles checkAndApplyStrategy(RewriteDataFiles action, String strategy, SortOrder sortOrder) {
+    if (strategy != null) {
+      if (strategy.equalsIgnoreCase("binpack")) {
+        RewriteDataFiles rewriteDataFiles = action.binPack();
+        if (sortOrder != null) {
+          // calling below method to throw the error as user has set both binpack strategy and sortorder
+          return rewriteDataFiles.sort(sortOrder);
+        }
+        return rewriteDataFiles;
+      } else if (strategy.equalsIgnoreCase("sort")) {
+        if (sortOrder == null) {
+          return action.sort();
+        }
+        return action.sort(sortOrder);
+      } else {
+        throw new IllegalArgumentException("unsupported strategy: " + strategy + ". Only binpack,sort is supported");
+      }
+    } else {
+      // overerride default binpack to sort, as startgey is not set but sort order is set.
+      if (sortOrder != null) {
+        return action.sort(sortOrder);
+      }
+    }
+    return action;
+  }
+
+  private SortOrder collectSortOrders(Table table, String sortOrderStr) {
+    if (sortOrderStr != null) {
+      String prefix = "ALTER TABLE temp WRITE ORDERED BY ";
+      try {
+        // Note: Reusing the existing Iceberg sql parser to avoid implementing the custom parser for sort orders.
```

Comment:
I think this is a cool workaround but a bit roundabout. I'm not sure I have a better solution though ...

Reply:
yeah. Thanks. 
I too didn't find any shorter way to do the same. Already spent quite some time on it :)